samples/javascript/index.js

```diff
@@ -0,0 +1,253 @@
+
+                        if (this.allHandledEventTypes.has(eventType)) {
+                            buildEventHandlers[eventType].forEach(handler => handler[`on${eventType}`](buildEventPayload));
+                        }
+                    }
+                }
+            ],
+            retry: {
+                interval: 2000,
+                maxRetries: 100
+            }
+        });
+    }
+
+    finishedProcessingBuild() {
+        this.numBuildsInProcess--;
+        setTimeout(() => this.processPendingBuilds(), 0); // process the next set of pending builds, if any
+    }
+}
+
+// Code below is a wrapper of EventSourcePolyfill to provide an oncomplete callback,  retry interval and max retries configuration
+function createServerSideEventStream(url, configuration) {
+    const STATUS_COMPLETE = 204;
+
+    let stream;
+    let retries;
+
+    const _onopen = configuration.onopen || noop;
+    const _onerror = configuration.onerror || noop;
+    const _oncomplete = configuration.oncomplete || noop;
+    const _configurationRetry = configuration.retry || { }
+    const _maxRetries = _configurationRetry.maxRetries || 3;
+    const _retryInterval = _configurationRetry.interval || 1000;
+
+    stream = createStream()
+
+    function createStream() {
+        stream = new EventSourcePolyfill(url, { headers: {'Authorization': `Basic ${BASIC_AUTH_TOKEN}`}});
+
+        stream.reconnectInterval = _retryInterval;
+
+        stream.onopen = (event) => {
+            retries = 0;
+            _onopen(event)
+        }
+
+        stream.onerror = (event) => {
+            // The server will send a 204 status code when the stream has finished sending events.
+            // The browser default EventSource implementation handles this use case as an error.
+            // We therefore map this from the error to the oncomplete callback for improved usage
+            if(event.status === STATUS_COMPLETE) {
+                _oncomplete()
+                return
+            }
+
+            // On all other errors, except the above handled complete event, the EventSourcePolyfill tries to reconnect
+            // to the server until it succeeds. To not do this indefinitely, we abort the reconnection loop if the specified _maxRetries limit is reached.
+            if(stream.readyState === EventSourcePolyfill.CONNECTING) {
+                if(_maxRetries > 0 && retries < _maxRetries) {
+                    // on failed events we get two errors, one with a proper
+                    // status and an undefined one, ignore the undefined to increase retry count correctly
+                    if(event.status != null) retries++;
+
+                } else {
+                    stream.close()
+                    console.log(`Connecting to ${url} ERROR: max retries reached ${_maxRetries}`);
+                }
+            }
+
+            _onerror(event)
+        }
+
+        addStreamEventListeners()
```

Comment:
I would inline this function call as it seems that is called only once and it is very short.